File: workplace/Assign4/src/assign4/Item.java
package assign4;
import java.text.NumberFormat;

//   Represents an item in a shopping cart.
public class Item {

	private String name;	// name of the item
	private double price;	// cost of the item
	private int quantity;	// how many of the item
	
	//  Create a new item with the given attributes.
	public Item (String itemName, double itemPrice, int numPurchased) {
		name = itemName;
		price = itemPrice;
		quantity = numPurchased;
	}
	
	//   Return a string with the information about the item
	public String toString () {
		NumberFormat fmt = NumberFormat.getCurrencyInstance();
		String item;

		item = name + (name.length() >= 8 ? "\t" : "\t\t");
		return (item + "   " + fmt.format(price) + "\t    " + quantity 
			+ "\t\t" + fmt.format(price * quantity));
	}
	
	//   Returns the unit price of the item
	public double getPrice() {
		return price;
	}
	
	//   Returns the name of the item
	public String getName() {
		return name;
	}
	
	//   Returns the quantity of the item
	public int getQuantity() {
		return quantity;
	}   
}

File: workplace/Assign4/src/assign4/Item.go
package main;

import "fmt"

//   Represents an item in a shopping cart.
type Item struct {
	name string;	// name of the item
	price float64;	// cost of the item
	quantity int;	// how many of the item
}
	
	//  Create a new item with the given attributes.
func NewItem (itemName string, itemPrice float64, numPurchased int) *Item {
	return &Item{itemName, itemPrice, numPurchased}
}
	
//   Return a string with the information about the item
func (I *Item) String () string {
	var item string;

	item = I.name + "\t"
	if (len(I.name) >= 8) {
		item += "\t" 
	} 
	return item + "   " + CurrencyFormat(I.price) + "\t    " + fmt.Sprintf("%d", I.quantity) +
		"\t\t" + CurrencyFormat(I.price * float64(I.quantity));
}

//   Returns the unit price of the item
func (I *Item) GetPrice() float64 {
	return I.price;
}

//   Returns the name of the item
func (I *Item) GetName() string {
	return I.name;
}

//   Returns the quantity of the item
func (I *Item) GetQuantity() int {
	return I.quantity;
}   
